scvi/external/cell2location/_base.py

```diff
@@ -0,0 +1,1060 @@
+                data_transform = self.data_transform_clusters()
+            elif data_transform == "log1p":
+                # use simple log1p transform
+                data_transform = torch.log1p
+                n_in = self.model.n_vars
+            elif (
+                isinstance(data_transform, dict)
+                and "var_std" in list(data_transform.keys())
+                and "var_mean" in list(data_transform.keys())
+            ):
+                # use data transform by scaling
+                n_in = self.model.n_vars
+                self.register_buffer(
+                    "var_mean",
+                    torch.tensor(
+                        data_transform["var_mean"].astype("float32").reshape((1, n_in))
+                    ),
+                )
+                self.register_buffer(
+                    "var_std",
+                    torch.tensor(
+                        data_transform["var_std"].astype("float32").reshape((1, n_in))
+                    ),
+                )
+                data_transform = self.data_transform_scale()
+            else:
+                # use custom data transform
+                data_transform = data_transform
+                n_in = self.model.n_vars
+
+            self._guide.append(
+                AutoNormalEncoder(
+                    pyro.poutine.block(
+                        self.model, expose=list(amortised_vars["sites"].keys())
+                    ),
+                    amortised_plate_sites=amortised_vars,
+                    n_in=n_in,
+                    n_hidden=n_hidden,
+                    data_transform=data_transform,
+                    encoder_kwargs=encoder_kwargs,
+                    single_encoder=single_encoder,
+                )
+            )
+
+        self._get_fn_args_from_batch = self._model._get_fn_args_from_batch
+
+    @property
+    def model(self):
+        return self._model
+
+    @property
+    def guide(self):
+        return self._guide
+
+    @property
+    def is_amortised(self):
+        return self._amortised
+
+    def data_transform_clusters(self):
+        def _data_transform(x):
+            return torch.log1p(torch.cat([x, x @ self.gene_clusters], dim=1))
+
+        return _data_transform
+
+    def data_transform_scale(self):
+        def _data_transform(x):
+            # return (x - self.var_mean) / self.var_std
+            return x / self.var_std
+
+        return _data_transform
+
+
+class TrainSampleMixin:
+    """
+    Reimplementation of cell2location [Kleshchevnikov20]_ model. This mixin class provides methods for:
+
+    - training models using minibatches (standard scVI interface) and using full data (copies data to GPU only once).
+    - computing median and quantiles of the posterior distribution using both direct and amortised inference
+    - generating samples from posterior distribution
+
+    """
+
+    @property
+    def _plan_class(self):
+        return PyroTrainingPlan
```

Comment:
this shouldn't be necessary if you rebase your branch on the head of master

Reply:
@adamgayoso merged the most recent master branch into this branch, does this solve it?